misc/toml-parser/src/main/java/sementic/nodes/TomlKey.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package sementic.nodes;
+
+import syntax.tree.SyntaxKind;
+
+/**
+ * Represents Key in TOML AST.
+ */
+public class TomlKey extends TomlNode {
+
+    public String name;
+    public SyntaxKind type;
```

Comment:
Since every node has a kind associated with it, shall we introduce the `kind` field to the abstract `TomlNode` class, and remove it from all other classes?